### PR TITLE
Integrated PEframe via optional docker based API service

### PR DESCRIPTION
**Pull request by Eshaan7**
#### Issue:
Closes Issue #6.

#### Test Run 
with pyintelowl: https://i.imgur.com/vTOwFJT.jpg

#### Changes in existing files:
- `PEframe_Scan_File` entry in `analyzers_configuration.json` and `Usage.md`
- tests in `test_files.py` via mock connection requests
- Celery task for peframe in `tasks.py`
- Documented how to of "Adding docker based integration" in `Contribute.md`.
- Documented how to use optional integrations and configuration of new environment files in `Installation.md`.

#### New Files:
- **`peframe.py`**: calls the API for analysis and saves result
- **`.env_template`:** template file to specify `COMPOSE_FILE` variable needed to run optional integrations.
- **`.env_file_integrations_template`:** template file for environment variables needed by optional integrations.
- **`integrations/`**: a folder to keep source code and docker files of optional integrations.
> Note: Entries for same have been added to `.gitignore`

PS: Only thing is remaining is adding entry for logfile in `generic_logs`.

### CODE REVIEW

File: integrations/peframe/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM python:3.8.2-alpine3.11
+
+MAINTAINER eshaan7bansal@gmail.com
+
+RUN apk update && apk upgrade
+RUN apk add --no-cache git libssl1.1 swig g++ gcc make openssl-dev python3-dev libffi-dev libmagic 	
+
+# Build PEframe
+RUN git clone https://github.com/guelfoweb/peframe.git
+WORKDIR peframe/
```
Comment (mlodic) [inline integrations/peframe/Dockerfile:10]: to easily understand where we can find the "additional" content of the container, please put peframe and the flask app in the same directory like `/opt/intel_owl` then chown it.

File: integrations/docker-compose.peframe.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    restart: unless-stopped
+    expose:
+      - "4000"
+    env_file:
+      - env_file_integrations
+    volumes:
+      - generic_logs:/var/log/intel_owl
+    depends_on:
+      - uwsgi
+    links:
```
[the remark this answers is not in the document]
Comment (mlodic) [reply integrations/docker-compose.peframe.yml:23]: `links` clause

File: api_app/script_analyzers/file_analyzers/peframe.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        r = requests.post("http://peframe:4000/run_analysis", files=files)
+        r_data = r.json()
+        if r.status_code==200:
+            max_tries = additional_config_params.get('max_tries', 15)
+            res = _poll_for_result(job_id, r_data['md5'], max_tries)
+        else:
+            raise AnalyzerRunException(r_data['error'])
+
+        results = {}
+        results['data'] = res
```
[the remark this answers is not in the document]
Comment (Eshaan7) [reply api_app/script_analyzers/file_analyzers/peframe.py:32]: Do you mean just do `results = res`  ?

[the remark this answers is not in the document]
Comment (mlodic) [reply api_app/script_analyzers/file_analyzers/peframe.py:32]: yep

**Comment by Eshaan7** [comment]
UPDATE:

- In Flask app, `SECRET_KEY`, if not specified in env variables would be generated randomly itself.
- PEframe process wasn't executing correctly. Fixed.
- Typos fixed
- created `docker-compose-for-tests.peframe.yml`
- Documentation updated as per pointed out in PR review

**Comment by mlodic** [comment]
Thank you. I made some refactoring to allow the logs to work properly.

**Pull request merged**